File: ml/python/base_model.py
class BaseModel:
    def __init__(self, config=None):
        self.config = config or {}
        self.model = None
        self.is_loaded = False

    def load(self):
        """Load the model"""
        raise NotImplementedError("Method 'load' must be implemented")

    def predict(self, input_data):
        """Run prediction on input"""
        raise NotImplementedError("Method 'predict' must be implemented")

    def dispose(self):
        """Dispose of model resources"""
        self.model = None
        self.is_loaded = False

File: backend/ml/js/BaseModel.js
/**
 * Base class for all ML models
 */
class BaseModel {
    constructor(config = {}) {
        this.config = config;
        this.model = null;
        this.isLoaded = false;
    }

    /**
     * Load the model
     * @returns {Promise<boolean>}
     */
    async load() {
        throw new Error("Method 'load' must be implemented");
    }

    /**
     * Run prediction on input
     * @param {any} input 
     * @returns {Promise<any>}
     */
    async predict(input) {
        throw new Error("Method 'predict' must be implemented");
    }

    /**
     * Dispose of model resources
     */
    dispose() {
        this.model = null;
        this.isLoaded = false;
    }
}

export default BaseModel;
